### ISSUE TITLE
Throttle Toggl API calls

### ISSUE THREAD

**Issue by consulthys** [mikerourke/toggl-to-clockify#1]
First of all, thanks for this awesome project!

One thing that needs to be taken care of is to add some throttling between calls to the Toggl API. If one has several pages of data the following error will be thrown when paging through Toggl entries:
```
(node:49735) UnhandledPromiseRejectionWarning: FetchError: invalid json response body at https://toggl.com/reports/api/v2/details?workspace_id=123456&user_agent=john%40doe.com&page=29&since=2018-01-01&until=2018-10-15 reason: Unexpected token < in JSON at position 0
    at /Users/consulthys/workspaces/toggl-to-clockify/node_modules/node-fetch/lib/index.js:239:32
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:49735) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:49735) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This is due to the fact that the Toggl API will return the following response if requests are [not properly throttled](https://github.com/toggl/toggl_api_docs#the-api-format):
```
<html>
<head><title>429 Too Many Requests</title></head>
<body bgcolor="white">
<center><h1>429 Too Many Requests</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

We can see on [lines 134-138 of `Toggl.ts`](https://github.com/mikerourke/toggl-to-clockify/blob/master/src/tools/Toggl.ts#L134-L138) that all fetch calls go out at the same time, so it might be ok if one has only very few pages, but not otherwise.

Basically, the same way you did for the calls to the Clockify API using the [`pause` function](https://github.com/mikerourke/toggl-to-clockify/blob/master/src/tools/Clockify.ts#L90-L97) would be perfect.

**Comment by mikerourke** [comment]
@consulthys I'll definitely add that in. I'm currently in the process of switching it over to a CLI app (rather than configuring locally). I haven't published to npm yet either. I accommodate for rate limits when adding Clockify entries, just not Toggl.

**Comment by consulthys** [comment]
Thanks for having implemented this in your last commit f2c84a34beefe28a823306cc6351e50b7761735f, it works like a charm.
